### PR TITLE
[extractor/audioboom] Sanitize webpage URL before downloading

### PR DESCRIPTION
**Pull request by tpikonen**
Fix JSON extraction and `uploader_url`  extraction.

When given a direct mp3 URL this extractor currently downloads the mp3 file as a webpage to be scraped and fails.
Sanitize the URL before downloading to be of type `https://audioboom.com/posts/<video_id>` to get the correct page.

I get random ads inserted into the beginning of the downloaded mp3 file, so the md5 tests fail.

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)

### CODE REVIEW

**Comment by pukkandan** [comment]
Add test

**Comment by pukkandan** [comment]
_VALID_URL allows `/boos/` too. How does this handle that?

**Comment by tpikonen** [comment]
I could not find an example of  `audioboom.com/boos' URL which would not redirect to `audioboom.com/posts`, so I'd say probably. But an example and a test case would be nice to have.

**Comment by pukkandan** [comment]
Could you add a test?

**Comment by tpikonen** [comment]
If you can find a 'boos' URL in the wild then sure, I haven't been able to find one. The ones listed [here](https://audioboom.com/tag/boos) are all 'no longer available' and the links are to 'posts' URLs anyway. I think boos are feature which was abandoned already a while ago.

File: yt_dlp/extractor/audioboom.py [hunk 1]
```diff
@@ -27,14 +40,15 @@ class AudioBoomIE(InfoExtractor):
...
         clip = None
 
         clip_store = self._parse_json(
-            self._html_search_regex(
-                r'data-new-clip-store=(["\'])(?P<json>{.+?})\1',
-                webpage, 'clip store', default='{}', group='json'),
+            unescapeHTML(self._html_search_regex(
+                r'data-react-class="V5DetailPagePlayer" *data-react-props=(["\'])(?P<json>{.+?})\1',
+                webpage, 'clip store', default='{}', group='json')),
             video_id, fatal=False)
```
Comment (pukkandan) [inline yt_dlp/extractor/audioboom.py:52]: Use self._search_json

File: yt_dlp/extractor/audioboom.py [hunk 2]
```diff
@@ -27,14 +40,15 @@ class AudioBoomIE(InfoExtractor):
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
-        webpage = self._download_webpage(url, video_id)
+        webpage = self._download_webpage(
+            'https://audioboom.com/posts/' + video_id, video_id)
```
Comment (pukkandan) [inline yt_dlp/extractor/audioboom.py:44]:
```suggestion
        webpage = self._download_webpage(f'https://audioboom.com/posts/{video_id}', video_id)
```

File: yt_dlp/extractor/audioboom.py [hunk 3]
```diff
@@ -27,14 +40,15 @@ class AudioBoomIE(InfoExtractor):
...
 
         clip = None
 
         clip_store = self._parse_json(
-            self._html_search_regex(
-                r'data-new-clip-store=(["\'])(?P<json>{.+?})\1',
-                webpage, 'clip store', default='{}', group='json'),
+            unescapeHTML(self._html_search_regex(
+                r'data-react-class="V5DetailPagePlayer" *data-react-props=(["\'])(?P<json>{.+?})\1',
```
Comment (pukkandan) [inline yt_dlp/extractor/audioboom.py:50]:
```suggestion
                r'data-react-class="V5DetailPagePlayer"\s*data-react-props=(["\'])(?P<json>{.+?})\1',
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/audioboom.py (hunk 1 above)
Comment (tpikonen) [reply yt_dlp/extractor/audioboom.py]: Direct JSON parsing from the regex result does not work, because the JSON is coded with HTML entities.

**Comment by tpikonen** [comment]
Anything to do here still?

**Pull request merged**